src/main/kotlin/org/jetbrains/research/ml/coding/assistant/system/finder/VertexFinders.kt

```diff
@@ -15,15 +15,25 @@ interface VertexFinder {
     fun findCorrespondingVertex(solutionSpace: SolutionSpace, partialSolution: PartialSolution): SolutionSpaceVertex?
 }
 
+/**
+ * Naive vertex finder returns the first vertex that matcher predicate `BooleanPartialSolutionMatcher.isMatched`
+ */
 class NaiveVertexFinder(override val matcher: BooleanPartialSolutionMatcher) : VertexFinder {
     override fun findCorrespondingVertex(
         solutionSpace: SolutionSpace,
         partialSolution: PartialSolution
     ): SolutionSpaceVertex? {
         return solutionSpace.graph.vertexSet().firstOrNull { matcher.isMatched(it, partialSolution) }
     }
+
+    override fun toString(): String {
+        return "NaiveVertexFinder(matcher=$matcher)"
+    }
```

Comment:
You can use this::class.simpleName and move this function out (create abstract class instead of an interface)